library/src/main/cpp/decoder_jxl.cpp

```diff
@@ -151,15 +137,54 @@ ImageInfo JpegxlDecoder::parseInfo() {
 }
 
 void JpegxlDecoder::decode(uint8_t* outPixels, Rect outRect, Rect inRect,
-                           bool rgb565, uint32_t sampleSize) {
-  JxlBasicInfo jxl_info;
-  std::vector<uint8_t> pixels =
-      decodeRGB(stream->bytes, stream->size, &jxl_info, 4);
+                           bool rgb565, uint32_t sampleSize,
+                           cmsHPROFILE target_profile) {
+  decode();
+
+  // Save transformed pixel data.
+  if (target_profile && mSrcProfile && !transformed) {
+    uint8_t* buf = pixels.data();
+
+    cmsUInt32Number inType;
+    std::vector<uint8_t> gray;
+    if (jxl_info.num_color_channels == 3) {
+      inType = TYPE_RGBA_8;
```

Comment:
Should it be assuming alpha?

Reply:
Yes. It's set to decode to 4 channels.
If you're curious (like I was) about the performance of libjxl with 3 channels and 4 channels for modular and vardct, it seemed to have a negligible difference in terms of resources and cpu usage. With 1 core, 2 megapixels, ~.2 seconds difference over 2000 iterations.